unit_tests/test_redis_aggregator.py

```diff
@@ -62,44 +96,98 @@ def fixture_mock_client_command_buffer(mocker):
+        with patch("gsy_e_sdk.redis_aggregator.GridFeeCalculation",
+                   return_value=grid_fee_calculation_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+            assert agg.grid_fee_calculation is grid_fee_calculation_mock
 
     @staticmethod
-    def test_constructor_redis_db_instantiated(aggregator):
-        assert aggregator.redis_db is not None
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_redis_db_instantiated():
+        redis_db_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.StrictRedis.from_url",
+                   return_value=redis_db_mock) as redis_from_url:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            redis_from_url.assert_called_with(LOCAL_REDIS_URL)
+            assert agg.redis_db is redis_db_mock
 
     @staticmethod
     def test_constructor_pubsub_instantiated(aggregator):
-        assert aggregator.pubsub is not None
+        aggregator.redis_db.pubsub.assert_called()
+        assert aggregator.pubsub is aggregator.redis_db.pubsub()
 
     @staticmethod
-    def test_constructor_client_command_buffer_instantiated(aggregator):
-        assert aggregator._client_command_buffer is not None
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_client_command_buffer_instantiated():
+        client_command_buffer_mock = MagicMock()
+        with patch("gsy_e_sdk.redis_aggregator.ClientCommandBuffer",
+                   return_value=client_command_buffer_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+        assert agg._client_command_buffer is client_command_buffer_mock
 
     @staticmethod
-    def test_constructor_threadpoolexecutor_instantiated(aggregator):
-        assert aggregator.executor is not None
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_threadpoolexecutor_instantiated():
+        ThreadPoolExecutor_mock = MagicMock
+        with patch("gsy_e_sdk.redis_aggregator.ThreadPoolExecutor",
+                   return_value=ThreadPoolExecutor_mock) as mocked_class:
+            agg = RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
+            mocked_class.assert_called()
+        assert agg.executor is ThreadPoolExecutor_mock
 
     @staticmethod
-    def test_constructor_connect_and_subscribe_called():
-        with patch("gsy_e_sdk.redis_aggregator."
-                   "RedisAggregator._connect_and_subscribe") as mocked_method:
-            RedisAggregator(aggregator_name=TEST_AGGREGATOR_NAME)
-            mocked_method.assert_called()
+    @pytest.mark.usefixtures("mock_transaction_id_and_timeout_blocking")
+    def test_constructor_connect_and_subscribe_side_effects_1():
```

Comment:
Are you planning on further splitting this method? I vote for doing this

Reply:
Addressed in [2b2611e](https://github.com/gridsingularity/gsy-e-sdk/pull/197/commits/2b2611e3b6fb37d7a1a217f1f8866b65660cbb2b). Thanks.